File: 1557. Minimum Number of Vertices to Reach All Nodes/Solution1557.cpp
#include <iostream>
#include <string>
#include <vector>
#include <algorithm>
using namespace std;

class Solution1557 {
public:
    vector<int> findSmallestSetOfVertices(int n, vector<vector<int>>& edges) {
        vector<int> res;
        vector<int> deg(n,0);
        for (vector<int> edge : edges) {
            deg[edge[1]]++;
        }
        for (int i = 0; i < n; i++) {
            if (deg[i] == 0) {
                res.push_back(i);
            }
        }
        return res;
    }
};

File: 1557. Minimum Number of Vertices to Reach All Nodes/Solution1557.py
from typing import List


class Solution1557:
    def findSmallestSetOfVertices(self, n: int, edges: List[List[int]]) -> List[int]:
        res = []
        deg = [0] * n
        for edge in edges:
            deg[edge[1]] += 1
        for i in range(n):
            if deg[i] == 0:
                res.append(i)
        return res